R/postprocess.R

```diff
@@ -124,34 +162,60 @@ enw_nowcast_summary <- function(fit, obs,
 #' @importFrom data.table setorderv
 #' @examples
 #' fit <- enw_example("nowcast")
-#' enw_nowcast_samples(fit$fit[[1]], fit$latest[[1]])
-enw_nowcast_samples <- function(fit, obs) {
+#' enw_nowcast_samples(fit$fit[[1]], fit$latest[[1]], fit$metamaxdelay[[1]])
+enw_nowcast_samples <- function(fit, obs, metamaxdelay = NULL) {
   nowcast <- fit$draws(
     variables = "pp_inf_obs",
     format = "draws_df"
   )
   nowcast <- coerce_dt(
-    nowcast, required_cols = c(".chain", ".iteration", ".draw")
+    nowcast,
+    required_cols = c(".chain", ".iteration", ".draw")
   )
   nowcast <- melt(
     nowcast,
     value.name = "sample", variable.name = "variable",
     id.vars = c(".chain", ".iteration", ".draw")
   )
-  max_delay <- nrow(nowcast) / (max(obs$.group) * max(nowcast$.draw))
+
```

Comment:
as this is duplicated across functions I think it could do with being an internal function to reduce technical debt.

Reply:
We also see a coverage decrease due to the duplicated code for checking the meta delay. Perhaps if we make that an internal as well and test it we can kill two birds with one stone?

Reply:
Maybe rather not, because the two lines are not identical, in the case of enw_nowcast_samples we also need to divide by the number of  draws, I don't really see why writing a function for two different one-liners that are quite verbatim